tests/tests-hz/src/main/java/com/hazelcast/simulator/tests/map/ComplexSqlPredicateTest.java

```diff
@@ -0,0 +1,545 @@
+     */
+    @TimeStep(prob = 0.4)
+    public void searchSingle(ThreadState state) {
+        String predicate = "id = '" + generatedIds[state.random.nextInt(generatedIds.length)] + "'";
+        System.out.println( map.values(new SqlPredicate(predicate)));
+    }
+
+    /**
+     * Should retrieve approx. half of the data set.
+     * @param state
+     */
+    @TimeStep(prob = 0.1)
+    public void searchHalf(ThreadState state) {
+        String predicate = "actionCode = " + (state.random.nextBoolean() ? ActionCode.CODE_0000.toCode() : ActionCode.CODE_9280.toCode());
+        map.values(new SqlPredicate(predicate));
+    }
+
+    /**
+     * Should retrieve approx. 25% of the data set.
+     * @param state
+     */
+    @TimeStep(prob = 0.1)
+    public void search25Percent(ThreadState state) {
+        String predicate = "panFirst6 < " + (panFirst6 / 2) + " AND panLast4 < " + (panLast4 / 2);
+        map.values(new SqlPredicate(predicate));
+    }
+
+    /**
+     * Should retrieve approx. 25% of the data set.
+     * @param state
+     */
+    @TimeStep(prob = 0.1)
+    public void search20Percent(ThreadState state) {
+        String predicate = "country = " + Country.values()[state.random.nextInt(Country.values().length)].toCode() + " OR country = " + Country.values()[state.random.nextInt(Country.values().length)].toCode();
+        map.values(new SqlPredicate(predicate));
+    }
+
+    /**
+     * Should retrieve approx. 16% of the data set.
+     * @param state
+     */
+    @TimeStep(prob = 0.1)
+    public void search16Percent(ThreadState state) {
+        String predicate = "cardBrand = " + CardBrand.values()[state.random.nextInt(CardBrand.values().length)].toCode() + " AND (currency = " + Currency.values()[state.random.nextInt(Currency.values().length)].toCode() + " OR currency = " + Currency.values()[state.random.nextInt(Currency.values().length)].toCode()  + ")";
+        map.values(new SqlPredicate(predicate));
+    }
+
+    /**
+     * Should retrieve approx. 10% of the data set.
+     * @param state
+     */
+    @TimeStep(prob = 0.1)
+    public void search10Percent(ThreadState state) {
+        String predicate = "country = " + Country.values()[state.random.nextInt(Country.values().length)].toCode();
+        map.values(new SqlPredicate(predicate));
+    }
+
+    /**
+     * Should retrieve approx. 4% of the data set.
+     * @param state
+     */
+    @TimeStep(prob = 0.1)
+    public void search4Percent(ThreadState state) {
+        String predicate = "lineOfBusiness = " + LineOfBusiness.values()[state.random.nextInt(LineOfBusiness.values().length)].toCode() + " AND transactionType = " + TransactionType.values()[state.random.nextInt(TransactionType.values().length)].toCode()
+                + " AND acquirerId = null AND currency = " + Currency.values()[state.random.nextInt(Currency.values().length)].toCode();
+        map.values(new SqlPredicate(predicate));
+    }
+
+    public class ThreadState extends BaseThreadState {
```

Comment:
Random is already available on BaseThreadState.

Reply:
Nice.